public/editor/scripts/project-list.js

```diff
@@ -81,44 +81,88 @@ require(["jquery", "constants", "analytics", "moment"], function($, Constants, a
-    // TODO: we can do better than this, but let's at least make it harder to lose data.
-    if(!window.confirm("{{ deleteProjectConfirmationText }}")) {
-      return false;
+    if($(".delete-button").length === 0){
+      var deleteBtn = document.createElement("div");
+      deleteBtn.appendChild(document.createTextNode("Delete Projects"));
+      deleteBtn.className = "delete-button";
+      deleteBtn.onclick = function() {
+        // TODO: we can do better than this, but let's at least make it harder to lose data.
+        if(!window.confirm("{{ deleteProjectConfirmationText }}")) {
+          return false;
+        }
+        
+        Array.prototype.forEach.call(projectsToDelete, function(project){
+          var projectId = project.attr("data-project-id");
+          var projectElementId = project.attr("id");
+          $("#" + projectElementId + " > .project-title").off("click");
+
+          analytics.event("DeleteProject");
+
+          var request = $.ajax({
+            headers: {
+              "X-Csrf-Token": $("meta[name='csrf-token']").attr("content")
+            },
+            type: "DELETE",
+            url: "/" + locale + "/projects/" + projectId,
+            timeout: Constants.AJAX_DEFAULT_TIMEOUT_MS
+          });
+          request.done(function() {
+            if(request.status !== 204) {
+              console.error("[Thimble error] sending delete request for project ", projectId, request.status);
+            }
+          });
+          request.fail(function(jqXHR, status, err) {
+            err = err || new Error("unknown network error");
+            console.error(err);
+          });
+
+          project.hide({
+            duration: 250,
+            easing: "linear",
+            done: function() {
+              project.remove();
+            }
+          });
+        });
+        projectsToDelete = [];
+        $(".delete-button").remove();
+      };
+      var listWrapper = document.getElementsByClassName("project-list-wrapper");
+      listWrapper[0].appendChild(deleteBtn);
     }
 
     var project = $(this).closest(".project");
-    var projectId = project.attr("data-project-id");
-    var projectElementId = project.attr("id");
-    $("#" + projectElementId + " > .project-title").off("click");
-
-    analytics.event({ category : analytics.eventCategories.PROJECT_ACTIONS, action : "Delete Project" });
-
-    var request = $.ajax({
-      headers: {
-        "X-Csrf-Token": $("meta[name='csrf-token']").attr("content")
-      },
-      type: "DELETE",
-      url: "/" + locale + "/projects/" + projectId,
-      timeout: Constants.AJAX_DEFAULT_TIMEOUT_MS
-    });
-    request.done(function() {
-      if(request.status !== 204) {
-        console.error("[Thimble error] sending delete request for project ", projectId, request.status);
+    var projectSelector = "#" + project.attr("id");
+    var deleteIndex = projectsToDelete.findIndex(function(x) {return x.attr("id") === project.attr("id");});
```

Comment:
I tried to use `indexOf` here originally but it does not seem to work when looking for objects in an object array. Since `findIndex` is not supported, the next best solution that would be compatible seems to be writing a simple loop to find the index manually. I used this implementation in my latest changes but if there is a better solution I'd be glad to change it.